doc/upgrades.md

```diff
@@ -16,27 +20,39 @@ To enable an upgrade in Hotshot protocol, it is essential to define the base ver
 These are defined in [NodeType implementation](../types/src/v0/mod.rs) for the Types (`SeqTypes` in our case).
 ```rust
 impl NodeType for SeqTypes {
-    type Base = StaticVersion<0, 1>;
-    type Upgrade = StaticVersion<0, 2>;
-    const UPGRADE_HASH: [u8; 32] = [
-        1, 0, 1, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0,
-    ], 
-    ..
+	type Base = StaticVersion<0, 1>;
+	type Upgrade = StaticVersion<0, 2>;
+	const UPGRADE_HASH: [u8; 32] = [
+		1, 0, 1, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0,
+	],
+	..
 }
 ```
 
-These parameters are fetched from the genesis TOML file and set in Hotshot config:
+Hotshot provides two modes for upgrades: view-based and time-based.
+
+View-based parameters allow nodes to vote and propose an upgrade based on the current node's view, while time-based parameters use Unix timestamps for the same purpose.
 
-- **start_voting_view:**  view at which voting for the upgrade proposal starts. In our implementation, this is set to 1 so that voting is enabled as soon as the node is started.
+To simplify configuration, these parameters are fetched from the genesis TOML file and set in the Hotshot config. The TOML file can include either view-based parameters or time-based parameters, but not both.
```

Comment:
Can the toml file contain view-based parameters for certain upgrades and some time-based parameters for other upgrades? Or do all the upgrades have to use use the same types of parameters?

Reply:
Yes, toml file has array of tables for upgrade so we define version, view/time parameters and the upgrade type separately for each upgrade. This can include different mode view or time based for upgrade

Reply:
I am going to add this in test once we have more upgrades

Reply:
One thing to note is that we don't support multiple upgrades, so view- or time-based settings need to be defined at the start in the hotshot config. Therefore, we just check what upgrade is planned and set those parameters, whether they are view-based or time-based.

Reply:
Ah yes remember that you mentioned we can only do one upgrade at a time. Thanks.